stl/inc/format

```diff
@@ -1871,13 +1870,13 @@ _NODISCARD _OutputIt _Fmt_write(_OutputIt _Out, const _CharT _Value) {
 template <class _CharT, class _OutputIt>
 _NODISCARD _OutputIt _Fmt_write(_OutputIt _Out, const void* const _Value) {
     // TRANSITION, Reusable buffer
-    array<char, _Format_min_buffer_length> _Buffer;
+    char _Buffer[_Format_min_buffer_length];
     const auto [_End, _Ec] =
-        _STD to_chars(_Buffer.data(), _Buffer.data() + _Buffer.size(), reinterpret_cast<uintptr_t>(_Value), 16);
+        _STD to_chars(_Buffer, _Buffer + _Format_min_buffer_length, reinterpret_cast<uintptr_t>(_Value), 16);

```

Comment:
Now that we don't have to deal with array's iterators not being T*s we can use `_STD begin()` and `_STD end()` here. 

This can sometimes prevent errors related to array decay into pointers, although it's not a problem here.